pi-manage

```diff
@@ -297,27 +298,28 @@ def create(directory="/var/lib/privacyidea/backup/",
     backup_file = "%s/%s-%s.tgz" % (directory, BASE_NAME, DATE)
 
     sqluri = app.config.get("SQLALCHEMY_DATABASE_URI")
-    sqltype = sqluri.split(":")[0]
+    parsed_sqluri = urlparse(app.config.get("SQLALCHEMY_DATABASE_URI"))
+    sqltype = parsed_sqluri.scheme
+
     if sqltype == "sqlite":
         productive_file = sqluri[len("sqlite:///"):]
         print("Backup SQLite %s" % productive_file)
         sqlfile = "%s/db-%s.sqlite" % (directory, DATE)
         call(["cp", productive_file, sqlfile])
     elif sqltype in MYSQL_DIALECTS:
-        m = re.match(r".*mysql://(.*):(.*)@(.*)/(\w*)\??(.*)", sqluri)
-        username = m.groups()[0]
-        password = m.groups()[1]
-        datahost = m.groups()[2]
-        database = m.groups()[3]
-        # We strip parameters, but we do not use them
-        _parameters = m.groups()[4]
+        username = parsed_sqluri.username
+        password = parsed_sqluri.password
+        hostname = parsed_sqluri.hostname
+        database = parsed_sqluri.path[1:]
         defaults_file = "{0!s}/mysql.cnf".format(conf_dir)
         _write_mysql_defaults(defaults_file, username, password)
-        call("mysqldump --defaults-file=%s -h %s %s > %s" % (
-            shlex_quote(defaults_file),
-            shlex_quote(datahost),
-            shlex_quote(database),
-            shlex_quote(sqlfile)), shell=True)
+        cmd = ['mysqldump', '--defaults-file={!s}'.format(
+            shlex_quote(defaults_file)), '-h',
+            shlex_quote(hostname)]
+        if parsed_sqluri.port:
+            cmd.extend(['-p', parsed_sqluri.port])
```

Comment:
The port number needs to be a string as well.
And somehow `mysqldump` doesn't work with these parameters. I'll have to check.

Reply:
fixed in cc453db88a442d312081d142089401aaf81a870a.